### PR TITLE
move mods normalizers into their own folder

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made?

Part of #2787 - move mods normalizer into its own folder for better organization...

No code actually changed -- it all just got moved around (so ignore all the deletion/addition).

Basically:

1. created a new "normalizers" folder under `app/services/cocina`
2. moved the `ModsNormalizer` class into this folder 
3. moved the `mods_normalizer` folder that all of the mods specific normalizers into this folder and renamed it `mods`
4. adjusted all the spots in the code for the new module names/paths
5. adjusted the spec locations and names to account for the moving
6. adjust rubocop paths 

## How was this change tested?

Existing test suite passes (and it failed spectacularly when i had typos/issues so the coverage actually seems to be good)

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -35,27 +54,27 @@ Lint/SuppressedException:
...
 # Configuration parameters: IgnoredMethods, CountRepeatedAttributes.
 Metrics/AbcSize:
   Max: 38
 
-# Offense count: 5
+# Offense count: 8
 # Configuration parameters: CountComments, CountAsOne.
 Metrics/ClassLength:
-  Max: 119
+  Max: 261
```
Comment (jcoyne) [inline .rubocop_todo.yml:65]: I really would prefer if this change didn't go in.  This raises the global threshold to something far higher than I would like. Can you restore the exception for the classes that brought this up?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
 # Note that changes in the inspected code, or installation of new
 # versions of RuboCop, may require this file to be generated again.
 
 # Offense count: 1
+# Cop supports --auto-correct.
+# Configuration parameters: EnforcedStyle, IndentationWidth.
+# SupportedStyles: aligned, indented, indented_relative_to_receiver
+Layout/MultilineMethodCallIndentation:
+  Exclude:
+    - 'app/services/cocina/normalizers/mods.rb'
```
Comment (jcoyne) [inline .rubocop_todo.yml:15]: Does this file still exist?

Comment (peetucket) [reply .rubocop_todo.yml:15]: oops relic of the changes that got out of sync...let me undo the rubocop config changes and the exceptions and see if I can target to just what i moved around

File: .rubocop_todo.yml (hunk 1 above)
Comment (peetucket) [reply .rubocop_todo.yml]: Reverted all the rubocop changes, which are now targeted to just the new class locations

**Review by jcoyne — APPROVED** [review]

**Pull request merged**